features/flows/itests/src/test/java/org/opennms/netmgt/flows/elastic/ThresholdingIT.java

```diff
@@ -0,0 +1,315 @@
+        "classpath:/META-INF/opennms/applicationContext-minimal-conf.xml",
+        "classpath:/META-INF/opennms/applicationContext-thresholding.xml",
+        "classpath:/META-INF/opennms/applicationContext-testPostgresBlobStore.xml",
+        "classpath:/META-INF/opennms/applicationContext-testThresholdingDaos.xml",
+        "classpath:/META-INF/opennms/applicationContext-testPollerConfigDaos.xml",
+        "classpath:/META-INF/opennms/applicationContext-rpc-utils.xml",
+})
+@JUnitConfigurationEnvironment
+@JUnitTemporaryDatabase(reuseDatabase = false, tempDbClass = MockDatabase.class)
+public class ThresholdingIT {
+    private static Logger LOG = LoggerFactory.getLogger(ThresholdingIT.class);
+
+    @Rule
+    public final ElasticSearchRule elasticSearchRule = new ElasticSearchRule();
+
+    @Autowired
+    private ThresholdingService thresholdingService;
+
+    @Autowired
+    private DatabasePopulator databasePopulator;
+
+    @Autowired
+    private InterfaceToNodeCache interfaceToNodeCache;
+
+    @Autowired
+    private TransactionTemplate transactionTemplate;
+
+    @Autowired
+    private OverrideableThresholdingDao thresholdingDao;
+
+    @Autowired
+    private OverrideableThreshdDao threshdDao;
+
+    @Autowired
+    private MockEventIpcManager mockEventIpcManager;
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Before
+    public void before() {
+        this.applicationContext.getAutowireCapableBeanFactory().createBean(DefaultResourceTypeMapper.class);
+
+        BeanUtils.assertAutowiring(this);
+
+        this.databasePopulator.populateDatabase();
+        this.interfaceToNodeCache.dataSourceSync();
+    }
+
+    private List<Flow> createMockedFlows(final int count) {
+        final List<Flow> flows = new ArrayList<>(count);
+        for (int i = 0; i < count; i++) {
+            final FlowDocument flowDocument = new FlowDocument();
+            flowDocument.setIpProtocolVersion(4);
+            flowDocument.setInputSnmp(10);
+            flowDocument.setOutputSnmp(20);
+            flowDocument.setSrcAddr(String.format("192.168.%d.%d", i % 256, 255 - (i % 256)));
+            flowDocument.setDstAddr(String.format("192.168.%d.%d", 255 - (i % 256), i % 256));
+            flowDocument.setSrcPort(1);
+            flowDocument.setDstPort(2);
+            flowDocument.setProtocol(6);
+            flowDocument.setBytes(1024L);
+            flowDocument.setDirection(Direction.INGRESS);
+
+            final TestFlow flow = new TestFlow(flowDocument);
+            flows.add(flow);
+        }
+        return flows;
+    }
+
+    private FlowRepository createFlowRepository(final JestClient jestClient) throws InterruptedException {
+        final var config = new ThresholdingConfig();
+        final var group = new Group();
+        group.setName("test");
+        group.setRrdRepository("/opt/opennms/share/rrd/snmp/");
+        final var threshold = new Threshold();
```

Comment:
Use a config file in src/test/resources instead?

Reply:
done